velox/exec/tests/DriverTest.cpp

```diff
@@ -1339,3 +1339,28 @@ DEBUG_ONLY_TEST_F(DriverTest, driverThreadContext) {
                   .assertResults("SELECT * FROM tmp");
   ASSERT_EQ(task.get(), capturedTask);
 }
+
+DEBUG_ONLY_TEST_F(DriverTest, nonReclaimableSection) {
+  // The driver framework will set non-reclaimable section flag when start
+  // executing operator method.
+  // Checks before getOutput method called.
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::Driver::runInternal::getOutput",
+      std::function<void(const exec::Values*)>([&](const exec::Values* values) {
+        ASSERT_FALSE(values->testingNonReclaimable());
+      }));
+  // Checks inside getOutput method execution.
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::Values::getOutput",
+      std::function<void(const exec::Values*)>([&](const exec::Values* values) {
+        ASSERT_TRUE(values->testingNonReclaimable());
+      }));
+
+  std::vector<RowVectorPtr> batches;
+  for (int i = 0; i < 2; ++i) {
+    batches.push_back(
+        makeRowVector({"c0"}, {makeFlatVector<int32_t>({1, 2, 3})}));
+  }
+  auto plan = PlanBuilder().values(batches).planNode();
+  AssertQueryBuilder(plan).copyResults(pool());
```

Comment:
maybe wrap in ASSERT_NO_THROW to clarity to the reader that you just want to make sure the query doesn't fail.